jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -2390,8 +2396,11 @@ static ClassLoader currentClassLoader() {
  * @since 1.5
  */
 public Annotation[] getAnnotations() {
-	Collection<Annotation> annotations = getAnnotationCache().annotationMap.values();
-	return annotations.toArray(new Annotation[annotations.size()]);
+	LinkedHashMap<Class<? extends Annotation>, Annotation> map = getAnnotationCache().annotationMap;
+	if (map != null) {
+		return map.values().toArray(EMPTY_ANNOTATION_ARRAY);
```

Comment:
Passing in the EMPTY_ANNOTATION_ARRAY may be less efficient than passing in an Annotation[] of the correct size. It means the code needs to call Array.newInstance() to create an array of the correct type and size.